projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -893,27 +936,34 @@ public function update_data( $request ) {
 					break;
 
 				case 'google_analytics_tracking_id':
-					$grouped_options = $grouped_options_current = (array) get_option( 'jetpack_wga' );
-					$grouped_options[ 'code' ] = $value;
+					$grouped_options_current = (array) get_option( 'jetpack_wga' );
+					$grouped_options         = $grouped_options_current;
+					$grouped_options['code'] = $value;
 
 					// If option value was the same, consider it done.
-					$updated = $grouped_options_current != $grouped_options ? update_option( 'jetpack_wga', $grouped_options ) : true;
+					$updated = $grouped_options_current !== $grouped_options
+						? update_option( 'jetpack_wga', $grouped_options )
+						: true;
 					break;
 
 				case 'dismiss_dash_app_card':
 				case 'dismiss_empty_stats_card':
 					// If option value was the same, consider it done.
-					$updated = get_option( $option ) != $value ? update_option( $option, (bool) $value ) : true;
+					$updated = get_option( $option ) !== $value ? update_option( $option, (bool) $value ) : true;
```

Comment:
I note `update_option()` will probably turn the bool into `'1'` or `''` when storing into the database. Should we keep the loose comparison here in case the caller is passing in bools or ints or `'0'`? Or do `$value = (string) (bool) $value;` first?

Reply:
👍 74547b3574d2da24d72b5cfb40b1068d3f945b3c